src/core/src/GltfUtil.cpp

```diff
@@ -317,20 +316,28 @@ const CesiumGltf::ImageCesium& getImageCesium(const CesiumGltf::Model& model, co
 }
 
 std::pair<std::string, uint64_t> parseAttributeName(const std::string& attributeName) {
-    const auto regex = std::regex("^([a-zA-Z_]*)(_([1-9]\\d*|0))?$");
-
-    std::string semantic;
-    std::string setIndex;
-
-    std::smatch matches;
+    int searchPosition = static_cast<int>(attributeName.size()) - 1;
+    int lastUnderscorePosition{-1};
+    while (searchPosition > 0) {
+        if (!isdigit(attributeName[searchPosition])) {
+            if (attributeName[searchPosition] == '_') {
+                lastUnderscorePosition = searchPosition;
+            }
 
-    if (std::regex_match(attributeName, matches, regex)) {
-        semantic = matches[1];
-        setIndex = matches[3];
+            break;
+        }
+        searchPosition--;
     }
 
-    uint64_t setIndexU64 = 0;
-    std::from_chars(setIndex.data(), setIndex.data() + setIndex.size(), setIndexU64);
+    std::string semantic{};
+    uint64_t setIndexU64{0};
+    if (lastUnderscorePosition == -1) {
+        semantic = attributeName;
+    } else {
+        semantic = attributeName.substr(0, lastUnderscorePosition);
+        auto setIndex = attributeName.substr(lastUnderscorePosition + 1);
+        std::from_chars(setIndex.data(), setIndex.data() + setIndex.size(), setIndexU64);
```

Comment:
It might be possible to avoid creating the `setIndex` string by passing in the offset / length to `std::from_chars`. One less heap allocation could help performance.

Reply:
Let's try that. It could only help.

Reply:
Without that heap allocation, it still levels out at 122 microseconds. It's probably a touch faster, but not enough to catch at the sampling rate. Still, let's keep it.

![image](https://github.com/CesiumGS/cesium-omniverse/assets/1040194/cd5116a2-decc-45f9-b96c-40a898f47d08)